docs/guides/cpp/protobuf.md

```diff
@@ -272,12 +167,125 @@ if (!message->ParseFromArray(static_cast<const void*>(it->message.data),
 std::cout << message->ShortDebugString() << std::endl;
 ```
 
-Finally, we close the reader.
+Finally, we close the reader:
 
 ```cpp
 reader.close();
 ```
 
+## Writing Protobuf
+
+Create an MCAP writer to start writing your Protobuf messages:
+
+```cpp
+mcap::McapWriter writer;
+mcap::McapWriterOptions opts("protobuf");
+auto s = writer.open("output.mcap");
+if (!s.ok) {
+  std::cerr << "Failed to open mcap writer: " << status.message << "\n";
+  throw std::runtime_error("could not open mcap writer");
+}
+```
+
+Configure it to your desired specifications using `McapWriterOptions`. For example, `opts.compressionLevel = mcap::CompressionLevel::Fast` will customize your writer to use a faster compression level.
+
+### Register schema
+
+Before we can write messages, we need to register a schema.
+
+You must use the fully-qualified name of the message type (e.g. `foxglove.PosesInFrame`) and provide a serialized `google::protobuf::FileDescriptorSet` for the schema itself. Generated Protobuf messages will contain enough information to reconstruct this `FileDescriptorSet` schema at runtime:
+
+```cpp
+// Recursively adds all `fd` dependencies to `fd_set`.
+void fdSetInternal(google::protobuf::FileDescriptorSet& fd_set,
+                   std::unordered_set<std::string>& files,
+                   const google::protobuf::FileDescriptor* fd) {
+  for (int i = 0; i < fd->dependency_count(); ++i) {
+    const auto* dep = fd->dependency(i);
+    auto [_, inserted] = files.insert(dep->name());
+    if (!inserted) continue;
+    fdSetInternal(fd_set, files, fd->dependency(i));
+  }
+  fd->CopyTo(fd_set.add_file());
+}
+
+// Returns a serialized google::protobuf::FileDescriptorSet containing
+// the necessary google::protobuf::FileDescriptor's to describe d.
+std::string fdSet(const google::protobuf::Descriptor* d) {
+  std::string res;
+  std::unordered_set<std::string> files;
+  google::protobuf::FileDescriptorSet fd_set;
+  fdSetInternal(fd_set, files, d->file());
+  return fd_set.SerializeAsString();
+}
+
+mcap::Schema createSchema(const google::protobuf::Descriptor* d) {
+  mcap::Schema schema(d->full_name(), "protobuf", fdSet(d));
+  return schema;
+}
+
+// Create a schema for the foxglove.PosesInFrame message.
+mcap::Schema path_schema = createSchema(foxglove::PosesInFrame::descriptor());
+writer.addSchema(path_schema);  // Assigned schema id is written to path_schema.id
+```
+
+### Register channel
+
+Next, we'll register a channel to write our messages to:
+
+```cpp
+mcap::Channel path_channel("/planner/path", "protobuf", path_schema.id);
+mcap.addChannel(path_channel);  // Assigned channel id written to path_channel.id
+```
+
+### Write messages
+
+We can now finally write messages to the channel via its id:
```

Comment:
my 2c: "via" sounds odd to me, but i'll leave it to your judgement.

Possibly capitalize ID?